art/attacks/evasion/imperceptible_asr/imperceptible_asr.py

```diff
@@ -108,16 +107,34 @@ def __init__(
         self._hop_size = masker.hop_size
         self._sample_rate = masker.sample_rate
 
-        # TensorFlow placeholders
-        self._delta = tf1.placeholder(tf1.float32, shape=[None, None], name="art_delta")
-        self._power_spectral_density_maximum_tf = tf1.placeholder(tf1.float32, shape=[None, None], name="art_psd_max")
-        self._masking_threshold_tf = tf1.placeholder(
-            tf1.float32, shape=[None, None, None], name="art_masking_threshold"
-        )
-        # TensorFlow loss gradient ops
-        self._loss_gradient_masking_threshold_op_tf = self._loss_gradient_masking_threshold_tf(
-            self._delta, self._power_spectral_density_maximum_tf, self._masking_threshold_tf
-        )
+        if isinstance(self.estimator, TensorFlowV2Estimator):
```

Comment:
I think to make this check reliable for custom estimators should we search for `TensorFlowV2Estimator` in `self.estimator.__mro__, e.g. if a user inherits from `TensorFlowLingvoAsr`?